### DISCUSSION TITLE
java.lang.RuntimeException: null error on join

### DISCUSSION THREAD

**Discussion by MinecraftCitiesNetwork** [EssentialsX/Essentials#4120] [Q&A] [answered]
Hi there. We are currently experiencing an issue where EssentialsX seems to show an error when players join back on our server, potentially having changed username or after a long hiatus and changed username.

Full Error displayed > https://paste.gg/p/anonymous/1190591bf6e4441a9d96f45a9cc7c82c

EssentialsX Version > [10:04:15 INFO]: You're running an experimental build of EssentialsX that is 1 builds behind the latest dev build!
I cannot change builds as I am currently needing to match versions for EssentialsXDiscord.

Server Version > [10:04:42 INFO]: This server is running Airplane version git-Airplane-"3ed988c" (MC: 1.16.5) (Implementing API ver

I would really appreciate some assistance.

**Comment by triagonal** [comment, marked as answer]
Despite the scary looking stacktrace, this particular warning isn't something that you need to worry about. If you're not experiencing issues related to users losing economy balances or irregular behaviour when running economy commands like `/balancetop` then you can safely ignore it. 

Like you said, this happens when players rejoin after having changed their username - Essentials is letting you know that it wasn't able to find the player by their new name, but was able to match their UUID to an existing player.

Recently the EssentialsX team added some fairly loud debug messages to certain Essentials economy operations to investigate issues where some plugins improperly accessed the Economy API and caused data loss/corruption. 

This message actually existed before the change, but the large stacktrace was added as part of the effort mentioned above (in order to see which plugin(s) initiated the operation). The need for the stracktraces should hopefully be removed with the introduction of our new Vault provider (#3975) which fixes the Economy data loss issues.

Edit 5/10/2020;
> Hello, this issue has been mitigated in recent EssentialsX dev builds!
>
> If you currently have damaged data, you can read more about the steps you can take to fix this and prevent further issues of this sort [here](https://github.com/EssentialsX/Essentials/issues/3956#issuecomment-779254544).
> 
> Download the latest dev build [here](https://essentialsx.net/downloads.html).

**Reply by MinecraftCitiesNetwork** [reply]
> Despite the scary looking stacktrace, this particular warning isn't something that you need to worry about. If you're not experiencing issues related to users losing economy balances or irregular behaviour when running economy commands like `/balancetop` then you can safely ignore it.
> 
> Like you said, this happens when players rejoin after having changed their username - Essentials is letting you know that it wasn't able to find the player by their new name, but was able to match their UUID to an existing player.
> 
> Recently the EssentialsX team added some fairly loud debug messages to certain Essentials economy operations to investigate issues where some plugins improperly accessed the Economy API and caused data loss/corruption.
> 
> This message actually existed before the change, but the large stacktrace was added as part of the effort mentioned above (in order to see which plugin(s) initiated the operation). The need for the stracktraces should hopefully be removed with the introduction of our new Vault provider (#3975) which fixes the Economy data loss issues.

I see, when should we roughly expect this new provider would you know?
